src/variable.jl

```diff
@@ -239,22 +239,29 @@ end
 
 function _construct_array_vars(macroname, var_name, type, call_args, val, prop, indices...)
     # TODO: just use Sym here
-    ndim = :($length(($(indices...),)))
+    #
+    # convert to Base.OneTo if possible
+    normalize = quote
+        idxs = ($(indices...),)
+        _idxs = map(i -> $(Base.OneTo)($last(i)), idxs)
```

Comment:
maybe check if the first is one? And fail if it's not.. We may want to  require it in the macro.

Reply:
One can still use OffsetArray if it fails.

Reply:
Also, checking the first is 1 doesn't check the range is `1:-1:-10` or `1:2:10`.

Reply:
Ok